### PR TITLE
Fix issue with `@overload` in projections

### LINKED ISSUE #2533: Type 'true' is not assignable to type 'boolean'
```tsp
import "@typespec/http";

using TypeSpec.Http;

@route("/test")
op test(@query p: boolean): void;

@overload(test)
op test_true(@query p: true): void;
```

https://cadlplayground.z22.web.core.windows.net/?c=aW1wb3J0ICJAdHlwZXNwZWMvaHR0cCI7Cgp1c2luZyBUeXBlU3BlYy5IdHRwOwoKQHJvdXRlKCIvdGVzdCIpCm9wIMQKKEBxdWVyeSBwOiBib29sZWFuKTogdm9pZMQzb3ZlcmxvYWQoxCrJM190cnVlyzjED8g1&e=%40typespec%2Fopenapi3&options=%7B%7D

`true` should be assignable to `boolean` in an overload. The point of this is to be able to define a root operation but overload it with separate logical operations that imply setting the parameter to a particular value.

### PR DESCRIPTION
**Pull request by tjprescott**
Fix #2533.

### CODE REVIEW

File: packages/compiler/src/core/checker.ts [hunk 1]
```diff
@@ -5253,6 +5253,10 @@ export function createChecker(program: Program): Checker {
       return source.kind === ReflectionNameToKind[target.name];
     }
 
+    if (source.kind === "Boolean" && target.kind === "Scalar" && target.name === "boolean") {
```
Comment (timotheeguerin) [inline packages/compiler/src/core/checker.ts:5256]: are you sure this is where the issue is, template constraint do validate that correctly already https://cadlplayground.z22.web.core.windows.net/?c=bW9kZWwgRm9vPFQgZXh0ZW5kcyBib29sZWFuPiB7CiAgZm9vOiBUCn0KCsYrVGVzdMgZVHJ1ZTrFPXRydWU%2BO8YWRmFsc8cXZsQLyBhJbnZhbGlkxhoiIj47Cn0%3D&e=%40typespec%2Fopenapi3&options=%7B%7D

Comment (timotheeguerin) [reply packages/compiler/src/core/checker.ts:5256]: it should already be doing that in the next function call 
image

Comment (tjprescott) [reply packages/compiler/src/core/checker.ts:5256]: This change _does_ fix the behavior.

Looks like there' s a bug in areScalarsRelated.

The first time it calls, it returns true, as expected.
image

But when it calls it again, with seemingly the same inputs, it returns false.
image

Comment (timotheeguerin) [reply packages/compiler/src/core/checker.ts:5256]: ah, could it be a projection issue? we are comparing types in different projector?

Comment (tjprescott) [reply packages/compiler/src/core/checker.ts:5256]: The only difference in the call stacks for the above cases is that the case where it unexpectedly returns false occurs during projection.

Comment (timotheeguerin) [reply packages/compiler/src/core/checker.ts:5256]: what is the stacktrace for the 2nd call?

Comment (timotheeguerin) [reply packages/compiler/src/core/checker.ts:5256]: my guess is that in the 2nd call we are passing a projected type and it compares the projected type from the non projected type which then fails

Comment (tjprescott) [reply packages/compiler/src/core/checker.ts:5256]: Yep, looks like during projection:
image

The reference equal on line 5239 evaluates to false even though the two are equivalent.

**Pull request merged**